// @think-herrmann/component-library/components/draggable-list/index.js
import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import {faSort} from '@fortawesome/fontawesome-free-solid'
import PropTypes from 'prop-types'
import Markdown from 'react-markdown/with-html'

import styles from './styles.css'
import {Colors} from '../colors';

/* This component pretty much lifted directly from https://github.com/siffogh/drag-and-drop-article/blob/master/src/App.js */

export default class DraggableList extends React.Component {

  constructor(props) {
    super()
    this.state = {
      items: props.list
    };
  }

  onDragStart = (e, index) => {
    this.draggedItem = this.state.items[index];
    e.dataTransfer.effectAllowed = "move";
    e.dataTransfer.setData("text/html", e.target.parentNode);
  };

  onDragOver = index => {
    const draggedOverItem = this.state.items[index];

    // if the item is dragged over itself, ignore
    if (this.draggedItem === draggedOverItem) {
      return;
    }

    // filter out the currently dragged item
    let items = this.state.items.filter(item => item !== this.draggedItem);

    // add the dragged item after the dragged over item
    items.splice(index, 0, this.draggedItem);

    this.setState({ items });
  };

  onDragEnd = () => {
    this.draggedIdx = null;
  };

  render() {
    return (
      <div className={styles.list}>
          <ul role="listbox">
            {this.state.items.map((item, idx) => (
              <li
                key={item}
                onDragOver={() => this.onDragOver(idx)}>
                <div
                  tabindex="0"
                  role="option"
                  data-draggable="item"
                  className={styles.drag}
                  draggable
                  onDragStart={e => this.onDragStart(e, idx)}
                  onDragEnd={this.onDragEnd}
                >
                <FontAwesomeIcon icon={["fas", "sort"]} className={styles.icon}/>
                </div>
                <span>{item}</span>
              </li>
            ))}
          </ul>
      </div>
    );
  }
}
