// @think-herrmann/component-library/brains/graphics/index.js
export {default as lincoln } from './Abraham Lincoln.svg'
export {default as aristotle } from './Aristotle.svg'
export {default as armstrong } from './Armstrong.svg'
export {default as bach } from './Bach.svg'
export {default as bigge } from './Bigge.png'
export {default as seattle } from './C Seattle.svg'
export {default as carter } from './Carter.png'
export {default as churchill } from './Churchill.svg'
export {default as chopin } from './Chopin.svg'
export {default as confucius } from './Confucius.svg'
export {default as corbusier } from './Corbusier.png'
export {default as curie } from './Curie.svg'
export {default as einstein } from './Einstein.svg'
export {default as gates } from './Gates.png'
export {default as geronimo } from './Geronimo.svg'
export {default as gandhi } from './Gandhi.svg'
export {default as henson } from './Henson.svg'
export {default as hoover } from './Hoover.svg'
export {default as jackson } from './Jackson.png'
export {default as kant } from './Kant.svg'
export {default as lagarde } from './Lagarde.svg'
export {default as lamb } from './Lamb.png'
export {default as lee } from './Lee.svg'
export {default as teresa } from './Mother Teresa.svg'
export {default as obama } from './Obama.png'
export {default as oprah } from './Oprah.svg'
export {default as picasso } from './Picasso.svg'
export {default as queen } from './Queen Mum.svg'
export {default as ramanujan } from './Ramanujan.png'
export {default as sousa } from './Sousa.png'
export {default as thatcher } from './Thatcher.svg'
export {default as vonnegut } from './Vonnegut.png'
export {default as walensa } from './Walensa.svg'
